applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.cpp

```diff
@@ -0,0 +1,1319 @@
+/***********************************************************************************/
+
+Vector SmallDisplacementMixedStrainDisplacementElement::GetBodyForce(
+    const GeometryType::IntegrationPointsArrayType& rIntegrationPoints,
+    const IndexType PointNumber) const
+{
+    KRATOS_TRY
+
+    const SizeType dim = GetGeometry().WorkingSpaceDimension();
+    const auto aux_body_force = StructuralMechanicsElementUtilities::GetBodyForce(*this, rIntegrationPoints, PointNumber);
+    Vector body_force(dim);
+    for (IndexType d = 0; d < dim; ++d) {
+        body_force(d) = aux_body_force(d);
+    }
+    return body_force;
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SmallDisplacementMixedStrainDisplacementElement::CalculateKinematicVariables(
+    KinematicVariables& rKinVariables,
+    const IndexType IP,
+    const GeometryType::IntegrationMethod& rIntegrationMethod) const
+{
+    KRATOS_TRY
+
+    const auto& r_geometry = GetGeometry();
+    const auto& r_integration_points = GetGeometry().IntegrationPoints(rIntegrationMethod);
+
+    // Shape functions
+    rKinVariables.N = r_geometry.ShapeFunctionsValues(rKinVariables.N, r_integration_points[IP].Coordinates());
+
+    // Calculate the inverse Jacobian
+    GeometryUtils::JacobianOnInitialConfiguration(r_geometry, r_integration_points[IP], rKinVariables.J0);
+    MathUtils<double>::InvertMatrix(rKinVariables.J0, rKinVariables.InvJ0, rKinVariables.detJ0);
+
+    KRATOS_ERROR_IF(rKinVariables.detJ0 < 0.0) << "Element ID: " << this->Id() << " is inverted. det(J0) = "
+        << rKinVariables.detJ0 << std::endl;
+
+    // Calculate the shape functions gradients
+    GeometryUtils::ShapeFunctionsGradients(r_geometry.ShapeFunctionsLocalGradients(rIntegrationMethod)[IP],
+        rKinVariables.InvJ0, rKinVariables.DN_DX);
+
+    // Compute B
+    CalculateB(rKinVariables.B, rKinVariables.DN_DX);
+
+    // Compute N_epsilon
+    CalculateN_EpsilonMatrix(rKinVariables.N_epsilon, rKinVariables.N);
+
+    // Compute the symmetric gradient of the displacements
+    noalias(rKinVariables.SymmGradientDispl) = prod(rKinVariables.B, rKinVariables.NodalDisplacements);
+
+    noalias(rKinVariables.NodalStrain) = prod(rKinVariables.N_epsilon, rKinVariables.NodalStrains);
+
+    // Calculate the equivalent total strain
+    CalculateEquivalentStrain(rKinVariables);
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SmallDisplacementMixedStrainDisplacementElement::CalculateB(
+    Matrix& rB,
+    const Matrix& rDN_DX
+    ) const
+{
+    KRATOS_TRY
+
+    StructuralMechanicsElementUtilities::CalculateB(*this, rDN_DX, rB);
+
+    KRATOS_CATCH("")
+}
```

Comment:
I think we can remove this function and directly call `StructuralMechanicsElementUtilities::CalculateB(*this, rDN_DX, rB);` wherever it's required.